### PR TITLE
Consolidate Import Page

### PR DESCRIPTION
**Pull request by bordoni**
_Ref:_ [C#38845](https://central.tri.be/issues/38845) + [C#38846](https://central.tri.be/issues/38846) + [C#38847](https://central.tri.be/issues/38847)
_Rel:_ moderntribe/events-facebook#33 moderntribe/events-eventbrite#81 moderntribe/events-importer-ical#37

### CODE REVIEW

File: src/io/csv/admin-views/general.php [hunk 1]
```diff
@@ -7,42 +7,20 @@
...
-
-	<h2><?php esc_html_e( 'Instructions', 'the-events-calendar' ); ?></h2>
+	<h1><?php esc_html_e( 'Instructions', 'the-events-calendar' ); ?></h1>
 	<p>
 		<?php printf( esc_html__( 'To import events, first select a %sDefault Import Event Status%s below to assign to your imported events.', 'the-events-calendar' ), '<strong>', '</strong>' ); ?>
 	</p>
 	<p>
 		<?php esc_html_e( 'Once your setting is saved, move to the applicable Import tab to select import specific criteria.', 'the-events-calendar' ); ?>
 	</p>
+	<?php do_action( 'tribe-import-general-infobox' ); ?>
```
Comment (barryhughes) [inline src/io/csv/admin-views/general.php]: Document this hook?

File: src/Tribe/Importer/Admin_Page.php [hunk 2]
```diff
@@ -20,6 +20,39 @@ public function register_admin_page() {
...
+			),
+		);
+		foreach ( $fields as $key => $field_args ) {
+			if ( 'imported_post_status[csv]' !== $key ){
+				$value = Tribe__Events__Main::getOption( $key, null );
+			} else {
+				$value = Tribe__Events__Importer__Options::get_default_post_status( 'csv' );
+			}
+
+			$field = new Tribe__Events__Field( $key, $field_args, $value );
```
Comment (barryhughes) [inline src/Tribe/Importer/Admin_Page.php]: `$field` is overwritten with each iteration of this loop and ultimately isn't retained when the method returns: do we need to keep a reference and can we set up a member field for that if we do?

File: src/Tribe/Importer/Options.php [hunk 3]
```diff
@@ -50,4 +51,34 @@ public static function getOption( $optionName, $default = '' ) {
...
+		if ( ! isset( $options[ $type ] ) ) {
+			$options[ $type ] = apply_filters( 'tribe-import-default_post_status_non_saved', 'publish', $type );
+		}
+
+		return apply_filters( 'tribe-import-default_post_status', $options[ $type ], $type );
+	}
+
+	public static function get_possible_stati() {
+		$stati = array(
+			'publish' => __( 'Published', 'tribe-events-ical-importer' ),
```
Comment (barryhughes) [inline src/Tribe/Importer/Options.php]: Text domain relates to iCal Importer - but we're in the core codebase here.

File: src/Tribe/Importer/Options.php [hunk 4]
```diff
@@ -50,4 +51,34 @@ public static function getOption( $optionName, $default = '' ) {
...
+		// A way to handle the legacy `imported_post_status`
+		if ( is_string( $options ) ) {
+			$options = array( $type => $options );
+		}
+
+		if ( ! isset( $options[ $type ] ) ) {
+			$options[ $type ] = apply_filters( 'tribe-import-default_post_status_non_saved', 'publish', $type );
+		}
+
+		return apply_filters( 'tribe-import-default_post_status', $options[ $type ], $type );
```
Comment (barryhughes) [inline src/Tribe/Importer/Options.php]: Curious about the mix of hyphens and underscores here ... is there a reason for that?

File: src/Tribe/Importer/Options.php [hunk 5]
```diff
@@ -50,4 +51,34 @@ public static function getOption( $optionName, $default = '' ) {
...
+	}
+
+	public static function get_possible_stati() {
+		$stati = array(
+			'publish' => __( 'Published', 'tribe-events-ical-importer' ),
+			'pending' => __( 'Pending', 'tribe-events-ical-importer' ),
+			'draft'   => __( 'Draft', 'tribe-events-ical-importer' ),
+		);
+
+		return apply_filters( 'tribe-import-possible_stati', $stati );
```
Comment (barryhughes) [inline src/Tribe/Importer/Options.php]: Similar to the above, interested in why we're mixing hyphens and underscores here.

File: src/Tribe/Importer/Plugin.php [hunk 6]
```diff
@@ -23,6 +24,8 @@ public static function initialize_admin() {
 		self::$admin = new Tribe__Events__Importer__Admin_Page();
 		add_action( 'admin_menu', array( self::$admin, 'register_admin_page' ) );
 		add_action( 'load-tribe_events_page_events-importer', array( self::$admin, 'handle_submission' ) );
+		add_action( 'tribe-import-general-settings', array( self::$admin, 'add_settings_fields' ) );
```
Comment (barryhughes) [inline src/Tribe/Importer/Plugin.php]: Hyphens vs underscores

File: src/Tribe/Importer/Plugin.php [hunk 7]
```diff
@@ -23,6 +24,8 @@ public static function initialize_admin() {
...
+
```
Comment (barryhughes) [inline src/Tribe/Importer/Plugin.php]: Whitespace

**Comment by bordoni** [comment]
I think the Hyphens Vs Underscore is a problem on the Importer in general, it started using Hyphens but our codebase normally uses Underscores...

**Comment by barryhughes** [comment]
With some special exceptions (like `base_name-unique_case`) WP itself generally uses underscores: if we're adding new hooks it feels like we should follow the same convention (which the rest of our own codebase also follows).

**Comment by bordoni** [comment]
Just replaced the usage of Dashes and Deprecated using `_doing_it_wrong`.

**Comment by barryhughes** [comment]
Nicely done :tada:

**Pull request merged**